### PR TITLE
feat(dev): Install postgresql on Apple M1 to build psycopg2-binary

### LINKED ISSUE #26465: psycogbinary2 does not install for Apple's M1
There's no wheel at the moment. I described that it can be produced in [here](https://github.com/psycopg/psycopg2/issues/1286).

### LINKED ISSUE #28606: Add `make prerequisites` target
Create a new target which installs different libraries via brew depending on the context.

For instance, on Apple M1 machines it should installed a couple more libraries.

### PR DESCRIPTION
**Pull request by armenzg**
Create a new target which installs different libraries via brew depending on the context.

For instance, on Apple M1 machines it should installed a couple more libraries.

In the future, the bootstrap script and documentation should be using this target.

Fixes #28606

### CODE REVIEW

File: requirements-base.txt [hunk 1]
```diff
@@ -35,7 +35,9 @@ Pillow==8.2.0; python_version == '3.6'
 Pillow==8.3.1; python_version > '3.6'
 progressbar2==3.32.0
 python-rapidjson==1.4
-psycopg2-binary==2.8.6
+psycopg2-binary==2.8.6; python_version == '3.6'
+# This removes needing to set CPPFLAGS & LDFLAGS
+psycopg2-binary==2.9.1; python_version > '3.8'
```
Comment (armenzg) [inline requirements-base.txt:40]: If I don't do this, I would have to do this somewhere in `install-py-dev`:

```shell
CPPFLAGS="-I/opt/homebrew/opt/openssl@1.1/include" \
    LDFLAGS="-L/opt/homebrew/opt/openssl@1.1/lib -L/opt/homebrew/opt/python@3.8/lib" \
    pip install psycopg2-binary==2.8.6
```

Comment (joshuarli) [reply requirements-base.txt:40]: You won't be installing this on 3.8.

```
Ignoring psycopg2-binary: markers 'python_version == "3.6"' don't match your environment
Ignoring psycopg2-binary: markers 'python_version > "3.8"' don't match your environment
```

Comment (armenzg) [reply requirements-base.txt:40]: Good catch! It was a last second commit that introduced the problem. I've corrected this and tested it out.

File: .github/workflows/development-environment.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
       fail-fast: false
     env:
       PIP_DISABLE_PIP_VERSION_CHECK: on
-      # Make the environment more similar to what Mac defaults to
+      # Make the environment more similar to hat Mac defaults to
```
Comment (billyvg) [inline .github/workflows/development-environment.yml:27]: mistake?

File: requirements-base.txt [hunk 3]
```diff
@@ -35,7 +35,9 @@ Pillow==8.2.0; python_version == '3.6'
 Pillow==8.3.1; python_version > '3.6'
 progressbar2==3.32.0
 python-rapidjson==1.4
-psycopg2-binary==2.8.6
+psycopg2-binary==2.8.6; python_version == '3.6'
+# This removes needing to set CPPFLAGS & LDFLAGS
```
Comment (billyvg) [inline requirements-base.txt:39]: Should we note that this only affects local development since we only support 3.6 in prod atm?

**Review by billyvg — APPROVED** [review]

**Review by joshuarli — APPROVED** [review]

**Pull request merged**